mongo/collection.go

```diff
@@ -936,7 +1026,13 @@ func (coll *Collection) Distinct(ctx context.Context, fieldName string, filter i
 	return retArray, replaceErrors(err)
 }
 
-// Find finds the documents matching a model.
+// Find executes a find operation (https://docs.mongodb.com/manual/reference/command/find/) and returns a Cursor over
+// the documents in the collection.
```

Comment:
I don't like the idea of mentioning the filter here because the previous docs PRs have also left it until the next paragraph. I changed this to "over a subset of the documents" rather than "over the documents" so it's clear that it will not necessarily return all documents.